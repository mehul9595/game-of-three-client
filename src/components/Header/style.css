
.App-logo {
    height: 2em;
    pointer-events: none;
  }
  
  .App-header {
    /* font-size: calc(10px + 1.5vmin); */
    color: white;
    display: flex;
    background: #50aadd;
    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
    
    height: 74px;
    align-items: center;
  }
  
  .App-header .title {
    width: 100%;
    margin: 5px 15px;
    display: flex;
    flex-direction: column;
    align-self: center;
  }
  
  .first-line {
    /* display: flex;
    align-self: flex-start; */
    font-size: calc(1em + 1vmin);
  }
  
  .sub-title {
    font-size: 1em;
    /* display: flex;
    align-self: flex-end; */
  }
  