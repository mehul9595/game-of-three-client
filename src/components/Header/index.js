import React from 'react';
import logo from "../../logo.svg";
import './style.css';

const Header = () => {
    return (
        <>
            <header className="App-header">
                <img src={logo} className="App-logo" alt="logo" />
                <div className="title">
                <div className="first-line">Scoober - Takeaway.com</div>
                <div className="sub-title">Win the game or win the job</div>
                </div>
            </header>
        </>
    );
}

export default Header;