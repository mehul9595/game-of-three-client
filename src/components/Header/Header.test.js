import React from "react";
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import Header from "./index";

describe("HeaderComponent", () => {
  afterEach(cleanup);
  test("should contain header", () => {
    const {getByRole} = render(<Header />);
    expect(getByRole("banner").className).toBe("App-header");
  });

  test("should contain logo image", () => {
    const { getByRole } = render(<Header />);

    expect(getByRole("img").attributes.getNamedItem("src").value).toBe("logo.svg");
  });

  test('should match the 1st line Title', () => {
    const { getByText } = render(<Header />);

    expect(getByText("Scoober - Takeaway.com")).toBeInTheDocument();
    expect(getByText("Scoober - Takeaway.com")).toHaveTextContent("Scoober - Takeaway.com");

  });

  test('should match the subtitle', () => {
    const { getByText } = render(<Header />);

    expect(getByText("Win the game or win the job")).toBeInTheDocument();
    expect(getByText("Win the game or win the job")).toHaveTextContent("Win the game or win the job");
  })
  
  

  test("should match the snapshot sucessfully", () => {
    const tree = render(<Header />);

    expect(tree.container.firstChild).toMatchSnapshot();
  });
});
