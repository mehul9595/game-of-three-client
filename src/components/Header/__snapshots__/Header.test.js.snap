// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should match the snapshot sucessfully 1`] = `
<header
  class="App-header"
>
  <img
    alt="logo"
    class="App-logo"
    src="logo.svg"
  />
  <div
    class="title"
  >
    <div
      class="first-line"
    >
      Scoober - Takeaway.com
    </div>
    <div
      class="sub-title"
    >
      Win the game or win the job
    </div>
  </div>
</header>
`;
